src/main.js

```diff
@@ -95,6 +96,26 @@ function getNLSConfiguration() {
 	return { locale: initialLocale, availableLanguages: {} };
 }
 
+function migrateUserDataDir(newDir) {
+	var oldDir = path.join(process.env.HOME, '.config', product.nameShort);
+	if (!fs.existsSync(oldDir)) {
+		return;
+	}
+	mkdirp(path.dirname(newDir));
+	fs.renameSync(oldDir, newDir);
```

Comment:
Please note that rename does not work if source and target are on different drives. I think we need to add a lot more error handling to your code so that in case of an issue we can show something useful to the user.